lib/jefferies_tube/capistrano/assets.rb

```diff
@@ -5,4 +5,30 @@
       download! "#{deploy_to}/shared/public/system", "public", recursive: true
     end
   end
+
+  namespace :s3 do
+    desc "Fetch s3 assets from AWS_BUCKET to public/system"
+    task :fetch do
+      puts "`aws` command not found. Please install the aws cli tools via `sudo pip install awscli`" and return unless File.executable?(`which aws`.strip)
+
+      appyml_text = nil
+      on roles(:web), primary: true do |host|
+        appyml_text = capture :cat, "#{deploy_to}/shared/config/application.yml"
+      end
+
+      appyml = YAML.load appyml_text
+      command = "AWS_ACCESS_KEY_ID=#{appyml['AWS_ACCESS_KEY_ID']} AWS_SECRET_ACCESS_KEY=#{appyml["AWS_SECRET_ACCESS_KEY"]} aws s3 sync s3://#{appyml['AWS_BUCKET']} public/system"
+      puts "Running: #{command}"
+
+      STDOUT.sync = true
+      require 'open3'
+      Open3.popen2e(command) do |i, stdout_and_err, thread|
```

Comment:
Does this work? it might be better to just fork the process and let it control the output, I ran it last week with habit and it think it was fancy like curses-style progress.  Especially since you don't need to do any cleanup or anything

Reply:
https://ruby-doc.org/core-2.4.1/Kernel.html#method-i-exec